src/test/java/com/volasoftware/tinder/services/implementations/AuthenticationServiceImplTest.java

```diff
@@ -120,27 +130,74 @@ void testCreatingAccountWhenEmailIsTakenThenExceptionIsThrown() {
     void testGettingAccountByEmailWhenGivenEmailExistsThenReturnActualAccount() {
         Account account = getAccount("alex", "t", "alex@gmail.com", "password", Gender.MALE);
         String email = "alex@gmail.com";
-        given(accountRepository.findOneByEmail(email)).willReturn(Optional.of(account));
+        given(authenticationRepository.findOneByEmail(email)).willReturn(Optional.of(account));
 
-        AccountDto accountFoundByEmail = accountService.getAccountByEmail(email);
+        AccountDto accountFoundByEmail = authenticationService.getAccountByEmail(email);
 
         assertEquals(email, accountFoundByEmail.getEmail());
     }
 
     @Test
     void testGettingAccountByEmailWhenGivenEmailNotExistsThenExceptionIsThrown() {
         String email = "phil@gmail.com";
-        given(accountRepository.findOneByEmail(email)).willThrow(new AccountNotFoundException("Account was not found"));
+        given(authenticationRepository.findOneByEmail(email)).willThrow(new AccountNotFoundException("Account was not found"));
 
         Exception exception = assertThrows(AccountNotFoundException.class,
-                () -> accountService.getAccountByEmail(email));
+                () -> authenticationService.getAccountByEmail(email));
 
         String expectedMessage = "Account was not found";
         String actualMessage = exception.getMessage();
 
         assertTrue(actualMessage.contains(expectedMessage));
     }
 
+    @Test
+    void testLoginWhenAccountNotFoundThenExceptionThrown() {
+        LoginRequest loginRequest = generateLoginRequest(null, null);
+
+        assertThrows(AccountNotFoundException.class, () -> authenticationService.login(loginRequest));
+    }
+
+    @Test
+    void testLoginWhenCredentialsAreWrongThenExceptionThrown() {
+        LoginRequest loginRequest = generateLoginRequest(EMAIL, PASSWORD);
```

Comment:
You can define a constant WRONG_PASSWORD, WRONG_EMAIL to clarify that we attempt to login with incorrect credentials. It will be more readable when look at the test case code.

Reply:
Applied suggested changes